File: JS/PGS_모의고사.js
function solution(answers) {
  var answer = [];

  let a = 0;
  let anum, bnum, cnum;
  let b = 0;
  let c = 0;
  let max = 0;
  for (let i = 0; i < answers.length; i++) {
    //1번 수포자
    anum = (i + 1) % 5;
    if (anum === 0) anum = 5;
    if (anum === answers[i]) a++;

    //2번 수포자
    if ((i + 1) % 2 === 1) {
      bnum = 2;
    } else {
      bnum = Math.floor((i + 1) / 2) % 4;
      if (bnum === 0) bnum = 4;
      if (bnum > 1) bnum += 1;
    }
    if (bnum === answers[i]) b++;

    //3번 수포자
    cnum = Math.floor((i % 10) / 2);
    if (cnum === 0) cnum = 3;
    else if (cnum === 1) cnum = 1;
    else if (cnum === 2) cnum = 2;
    else if (cnum === 3) cnum = 4;
    else if (cnum === 4) cnum = 5;

    if (cnum === answers[i]) c++;
  }
  max = Math.max(a, b, c);
  let list = [a, b, c];
  console.log(list);
  for (let i = 0; i < 3; i++) {
    if (list[i] === max) answer.push(i + 1);
  }

  return answer;
}

File: Python/PGS_모의고사.py
def solution(answers):
    answer = []
    answer_num = len(answers)
    cnt_list = []
    
    # 1수포자
    cnt1 = 0
    arr1 = [5,1,2,3,4]
    
    # 2수포자
    cnt2 = 0
    arr2 = [5,2,1,2,3,2,4,2]    
    
    # 3수포자
    cnt3 = 0
    arr3 = [5,3,3,1,1,2,2,4,4,5]    
    
    for i in range(1, answer_num+1):
        temp1 = arr1[i%5]
        if temp1 == answers[i-1]:
            cnt1 += 1
            
        temp2 = arr2[i%8]
        if temp2 == answers[i-1]:
            cnt2+=1
            
        temp3 = arr3[i%10]
        if temp3 == answers[i-1]:
            cnt3+=1
    
    cnt_list.extend([cnt1,cnt2,cnt3])
    
    result = max(max(cnt1,cnt2),cnt3)

    for i in range(0,3):
        if cnt_list[i] == result:
            answer.append(i+1)
            
    return answer